doc/help.md

```diff
@@ -580,14 +580,70 @@ PS1=$x.
 
 These builtins take input and output.  They're often used with redirects.
 
+<h4 id="read">read</h4>
+
+The read builtin reads lines from stdin into a variable ($REPLY by default)
+It also splits input according to the $IFS delimiter, and will assign fields
+to several variables if specified.
+
+  -a ARRAY  assign the splitted words to elements of an array
+  -d CHAR   use DELIM as delimiter, instead of newline
+  -e        (unimplemented) use readline to obtain the line
+  -i STR    (unimplemented) use STR as the initial text for readline
+  -n NUM    read NUM characters, but return before if delimiters are found
+  -N NUM    read up to NUM characters, ignoring delimiters.
+  -p STR    print the string PROMPT before reading input
+  -r        raw mode: don't let backslashes escape characters
+  -s        silent: do not echo input coming from a terminal
+  -t NUM    time out and fail after TIME seconds.
+  -u FD     read from file descriptor FD instead of the standard input
+
 <h4 id="echo">echo</h4>
 
 The echo builtin prints its arguments to stdout, separated by a space, and
-terminated by a newline.  TODO: Describe flags, etc.
-
-<h4 id="read">read</h4>
+terminated by a newline.
+
+  -e  enable interpretation of backslash escapes
+  -E  (unimplemented)
+  -n  omit the trailing newline
+
+Backslash sequences recognized by -e:
+  \\         backslash
+  \a         (unimplemented ?) alert (BEL)
+  \b         backspace
+  \c         stop processing remaining input
+  \e         escape next character
+  \f         form feed, equivalent to \n + 4 spaces
+  \n         new line
+  \r         carriage return, returns to the beggining of the line
+  \t         horizontal tab
+  \v         vertical tab
+  \0NNN      print character specified as an octal value with 1 to 3 octal
+             digits 
+  \xHH       print character specified as an hexadecimal value with 1 to 2
+             hex digits
+  \uHHHH     Unicode character specified as an hexadecimal value with 1 to
+             4 hex digits 
+  \UHHHHHHHH Unicode character specified as an hexadecimal value with 1 to
+             8 hex digits
+
+<h4 id="readarray">readarray</h4>
+
+Alias for mapfile
+
+<h4 id="mapfile">mapfile</h4>
+
+The mapfile builtin reads lines from stdin into an array
```

Comment:
Hm this only has `-t` implemented

https://github.com/oilshell/oil/blob/master/frontend/flag_def.py#L118

I would omit the (unimplemented) ones altogether (or somehow comment them out).  Because they might not be implemented for a long time!

Reply:
I have added a usage line and made sure that the options existed. I also have added a few more commands!